src/shared_realm.cpp

```diff
@@ -554,5 +555,110 @@ util::Optional<int> Realm::file_format_upgraded_from_version() const
     return util::none;
 }
 
+#define VERSION_CAST(version_id) {(version_id).version, (version_id).index}
```

Comment:
Ah, the problem is I cannot define a function that operates on `Realm::HandoverPackage::VersionID` since it's a private member. Any solutions? @tgoyne